src/libcore/iter/traits.rs

```diff
@@ -352,6 +352,13 @@ pub trait Extend<A> {
     fn extend<T: IntoIterator<Item=A>>(&mut self, iter: T);
 }
 
+#[stable(feature = "extend_for_unit", since = "1.27.0")]
+impl Extend<()> for () {
+    fn extend<T: IntoIterator<Item = ()>>(&mut self, iter: T) {
+        for _ in iter {}
```

Comment:
I believe `iter.for_each(drop)` can be faster for some iterators like chain.

Reply:
Huh, interesting.